File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int ans = 0;

        HashSet<Integer> set = new HashSet<>();

        for(int num : nums)
            set.add(num);


        for(int num : set){

            if(!set.contains(num - 1)){ 

                int curr = num;
                int len = 1;

                while(set.contains(curr + 1)){
                    curr++;
                    len++;
                }

                ans = Math.max(ans, len);
            }
        }

        return ans;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set()

        ans = 0

        for num in nums :
            s.add(num)

        for x in s :
            if (x - 1) not in s:
                curr = x
                len = 1

                while (curr + 1) in s:
                    len += 1
                    curr += 1

                ans = max(ans , len)

        return ans

